File: src/_02_StructuralPattern/_02_03_Composite/java/after/Item.java
package _02_StructuralPattern._02_03_Composite.java.after;

public class Item implements Component {

  private String name;
  private int price;

  public Item(String name, int price) {
    this.name = name;
    this.price = price;
  }

  // name
  public String getName() {
    return name;
  }
  
  @Override
  public int getPrice() {
    return price;
  }
  
}

File: src/_02_StructuralPattern/_02_03_Composite/Item.js
import Component from "./Component.js";

export default class Item extends Component {
  /** @type { string } */
  #name;

  /** @type { number } */
  #price;

  /**
   * @param { string } name 
   * @param { number } price 
   */
  constructor(name, price) {
    super();
    this.#name = name;
    this.#price = price;
  }

  /**
   * @override
   * @returns { number }
   */
  getPrice() {
    return this.#price;
  }
}